x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/repositories/DeleteInternalRepositoryAction.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ccr.action.repositories;
+
+import org.elasticsearch.action.Action;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.repositories.RepositoriesService;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+public class DeleteInternalRepositoryAction extends Action<ActionResponse> {
+
+    public static final DeleteInternalRepositoryAction INSTANCE = new DeleteInternalRepositoryAction();
+    public static final String NAME = "cluster:admin/internal_repository/delete";
+
+    private DeleteInternalRepositoryAction() {
+        super(NAME);
+    }
+
+    @Override
+    public ActionResponse newResponse() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Writeable.Reader<ActionResponse> getResponseReader() {
+        return in -> new ActionResponse() {};
```

Comment:
safer to use the `ActionResponse(StreamInput in)` constructor here. Maybe use add a dummy response sub-class here.